airbyte-integrations/connectors/destination-pubsub/src/test-integration/java/io/airbyte/integrations/destination/pubsub/PubsubDestinationAcceptanceTest.java

```diff
@@ -0,0 +1,250 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.pubsub;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.airbyte.integrations.destination.pubsub.PubsubDestination.CONFIG_CREDS;
+import static io.airbyte.integrations.destination.pubsub.PubsubDestination.CONFIG_PROJECT_ID;
+import static io.airbyte.integrations.destination.pubsub.PubsubDestination.CONFIG_TOPIC_ID;
+import static io.airbyte.integrations.destination.pubsub.PubsubDestination.NAMESPACE;
+import static io.airbyte.integrations.destination.pubsub.PubsubDestination.STREAM;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.api.client.util.Lists;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.cloud.pubsub.v1.SubscriptionAdminSettings;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminSettings;
+import com.google.cloud.pubsub.v1.stub.GrpcSubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStubSettings;
+import com.google.common.collect.ImmutableMap;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.standardtest.destination.DestinationAcceptanceTest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PubsubDestinationAcceptanceTest extends DestinationAcceptanceTest {
+
+  private static final Logger LOGGER = LoggerFactory
+      .getLogger(PubsubDestinationAcceptanceTest.class);
+  private static final Path CREDENTIALS_PATH = Path.of("secrets/credentials.json");
+  private TopicAdminClient topicAdminClient;
+  private SubscriptionAdminClient subscriptionAdminClient;
+  private TopicName topicName;
+  private ProjectSubscriptionName subscriptionName;
+  private Credentials credentials;
+  private JsonNode configJson;
+  // TODO: is this ok for a streaming destination test?
```

Comment:
yup this makes sense since we can't read the data multiple times -- you should just explain that in a comment 
```suggestion
  // Store retrieved data during the test run since we can't re-read it multiple times (ACKing messages causes them to be removed from pubsub)
```